packages/sidecar/src/main.ts

```diff
@@ -49,269 +54,145 @@ type SideCarOptions = {
-        res.setHeader(header, value);
-      }
-      if (req.method === "OPTIONS") {
-        res.writeHead(204, {
-          "Cache-Control": "no-cache",
-        });
-        res.end();
-        return;
-      }
-      return handler(req, res, pathname, searchParams);
-    },
-    { name: "enableCORS", op: "sidecar.http.middleware.cors" },
-  );
-
-const streamRequestHandler = (buffer: MessageBuffer<Payload>, incomingPayload?: IncomingPayloadCallback) => {
-  return function handleStreamRequest(
-    req: IncomingMessage,
-    res: ServerResponse,
-    pathname?: string,
-    searchParams?: URLSearchParams,
-  ): void {
-    if (
-      req.method === "GET" &&
-      req.headers.accept &&
-      req.headers.accept === "text/event-stream" &&
-      pathname === "/stream"
-    ) {
-      res.writeHead(200, {
-        "Content-Type": "text/event-stream",
-        "Cache-Control": "no-cache",
-        Connection: "keep-alive",
-      });
-      res.flushHeaders();
-      // Send something in the body to trigger the `open` event
-      // This is mostly for Firefox -- see getsentry/spotlight#376
-      res.write("\n");
-
-      const useBase64 = searchParams?.get("base64") != null;
-      const base64Indicator = useBase64 ? ";base64" : "";
+const streamGetRequestHandler = (buffer: MessageBuffer<Payload>): Handler => {
+  return function handleStreamRequest(c) {
+    // TODO: Check if we still need this
+    // // Send something in the body to trigger the `open` event
+    // // This is mostly for Firefox -- see getsentry/spotlight#376
+    // res.write("\n");
+    const useBase64 = c.req.query("base64") != null;
+    const base64Indicator = useBase64 ? ";base64" : "";
+    return streamSSE(c, async stream => {
       const dataWriter = useBase64
-        ? (data: Buffer) => res.write(`data:${data.toString("base64")}\n`)
+        ? (data: Buffer) => [`data:${data.toString("base64")}`]
         : (data: Buffer) => {
+            const _data = [];
             // The utf-8 encoding here is wrong and is a hack as we are
             // sending binary data as utf-8 over SSE which enforces utf-8
             // encoding. This is only for backwards compatibility
             for (const line of data.toString("utf-8").split("\n")) {
               // This is very important - SSE events are delimited by two newlines
-              res.write(`data:${line}\n`);
+              _data.push(`data:${line}`);
             }
+
+            return _data;
           };
+
       const sub = buffer.subscribe(([payloadType, data]) => {
         logger.debug("🕊️ sending to Spotlight");
-        res.write(`event:${payloadType}${base64Indicator}\n`);
-        dataWriter(data);
+        stream.writeln(`event:${payloadType}${base64Indicator}`);
+        for (const line of dataWriter(data)) {
+          stream.writeln(line);
+        }
         // This last \n is important as every message ends with an empty line in SSE
-        res.write("\n");
+        stream.write("\n");
```

Comment:
We should probably replace all these with `stream.writeSSE({})` if we are using Hono?